Flow.Launcher/Helper/HotKeyMapper.cs

```diff
@@ -0,0 +1,136 @@
+            }
+            catch (Exception)
+            {
+                string errorMsg =
+                    string.Format(InternationalizationManager.Instance.GetTranslation("registerHotkeyFailed"),
+                        hotkeyStr);
+                MessageBox.Show(errorMsg);
+            }
+        }
+
+        internal static void RemoveHotkey(string hotkeyStr)
+        {
+            if (!string.IsNullOrEmpty(hotkeyStr))
+            {
+                HotkeyManager.Current.Remove(hotkeyStr);
+            }
+        }
+
+        internal static void OnHotkey(object sender, HotkeyEventArgs e)
+        {
+            if (!ShouldIgnoreHotkeys())
+            {
+                UpdateLastQUeryMode();
+
+                mainViewModel.ToggleFlowLauncher();
+                e.Handled = true;
+            }
+        }
+
+        /// <summary>
+        /// Checks if Flow Launcher should ignore any hotkeys
+        /// </summary>
+        private static bool ShouldIgnoreHotkeys()
+        {
+            return settings.IgnoreHotkeysOnFullscreen && WindowsInteropHelper.IsWindowFullscreen();
+        }
+
+        private static void UpdateLastQUeryMode()
+        {
+            switch(settings.LastQueryMode)
+            {
+                case LastQueryMode.Empty:
+                    mainViewModel.ChangeQueryText(string.Empty);
+                    break;
+                case LastQueryMode.Preserved:
+                    mainViewModel.LastQuerySelected = true;
+                    break;
+                case LastQueryMode.Selected:
+                    mainViewModel.LastQuerySelected = false;
+                    break;
+                default:
+                    throw new ArgumentException($"wrong LastQueryMode: <{settings.LastQueryMode}>");
+
+            }
+        }
+
+        internal static void LoadCustomPluginHotkey()
+        {
+            if (settings.CustomPluginHotkeys == null)
+                return;
+
+            foreach (CustomPluginHotkey hotkey in settings.CustomPluginHotkeys)
+            {
+                SetCustomQueryHotkey(hotkey);
+            }
+        }
+
+        internal static void SetCustomQueryHotkey(CustomPluginHotkey hotkey)
+        {
+            SetHotkey(hotkey.Hotkey, (s, e) =>
+            {
+                if (ShouldIgnoreHotkeys())
+                    return;
+
+                mainViewModel.MainWindowVisibility = Visibility.Visible;
+                mainViewModel.ChangeQueryText(hotkey.ActionKeyword);
+            });
+        }
+
+        internal static bool CheckAvailability(HotkeyModel currentHotkey)
+        {
+            try
+            {
+                HotkeyManager.Current.AddOrReplace("HotkeyAvailabilityTest", currentHotkey.CharKey, currentHotkey.ModifierKeys, (sender, e) => { });
+
+                return true;
+            }
+            catch
+            {
+            }
```

Comment:
We can remove the catch

Reply:
cant remove it, catch is needed to go to `return false` at the bottom, you can just use catch but you duplicate code for no benefit, so may be just leave it?